routers/web/repo/actions/actions_test.go

```diff
@@ -154,3 +158,20 @@ func TestReadWorkflow_WorkflowDispatchConfig(t *testing.T) {
 		Type:        "boolean",
 	}, workflowDispatch.Inputs[2])
 }
+
+func Test_loadIsRefDeleted(t *testing.T) {
+	unittest.PrepareTestEnv(t)
+
+	runs, total, err := db.FindAndCount[actions_model.ActionRun](db.DefaultContext, actions_model.FindRunOptions{RepoID: 4})
+	assert.NoError(t, err)
+	assert.Len(t, runs, 3)
+	assert.EqualValues(t, 3, total)
+	for _, run := range runs {
+		assert.False(t, run.IsRefDeleted)
+	}
+
+	assert.NoError(t, loadIsRefDeleted(db.DefaultContext, 4, runs))
+	for _, run := range runs {
+		assert.True(t, run.IsRefDeleted)
+	}
```

Comment:
I won't block it but I can't understand what you are testing.

All are "deleted"? No survivor? Isn't it exactly the old bug's phenomenon?

Reply:
This is according to the test fixtures. Currently, there are only 3 action runs in the table and all of them belong to repo id 4, and no branch exists in the branch table.

Reply:
tests updated